src/braket/circuits/circuit.py

```diff
@@ -710,13 +710,15 @@ def apply_gate_noise(
         if not self.qubits:
             raise IndexError("Gate noise cannot be applied to an empty circuit.")
 
+        decomposed_circ = self.decompose()
```

Comment:
Why should this always decompose on application of noise ?

Reply:
Decomposition can also happen at execution time.
This allows for seeing what gates have noise applied and doesn't require tracking noise to be applied later to the QuantAlgo.